config.go

```diff
@@ -235,6 +235,16 @@ type Config struct {
 	// PreVoteDisabled deactivate the pre-vote feature when set to true
 	PreVoteDisabled bool
 
+	// FastRecovery controls if the Raft server should use the fast recovery
+	// mechanism. Fast recovery requires a LogStore implementation that
+	// support commit tracking. When such a store is used and this config
+	// enabled, raft nodes will replay all known-committed logs on disk
+	// before completing `NewRaft` on startup. This is mainly useful where
+	// the application allows relaxed-consistency reads from followers as it
+	// will reduce how far behind the follower's FSM is when it starts. If all reads
+	// are forwarded to the leader then there won't be observable benefit from this feature.
+	FastRecovery bool
```

Comment:
This is great point actually. I'm not sure "Fast" does capture the semantics in any case really: mostly startup will take marginally to a lot longer, but on the plus side the FSM will actually startup in the same state it was before the node restarted which is probably what most users of the library assumed was the case already!

> If my understanding is accurate

Yeah I think you perfectly described the tradeoff. I think `CommittedLogRecovery` is a reasonable name for the thing and we can explain the semantics in the comments.

Reply:
@schmichael  @banks How about `RecoveryCommitted`?

Reply:
thanks!

Reply:
Naming is the worst. :sweat_smile: Looking around a bit we're not totally consistent, but I think generally:
1. `Recovery` refers to remediating a failure of some kind ([`RecoverCluster`](https://pkg.go.dev/github.com/hashicorp/raft#RecoverCluster) and [many comments](https://github.com/search?q=repo%3Ahashicorp%2Fraft%20recover&type=code))
2. [`Restore`](https://pkg.go.dev/github.com/hashicorp/raft#Raft.Restore) is the term used for this process: replaying logs on startup.

So I think switching `Recovery -> Restore` is appropriate.

After that I think I prefer `RestoreCommittedLogs` a tiny bit over `RestoreCommitted` since `Committed` alone is a tiny bit ambiguous. I would accept either though!